merlin/core/compat/__init__.py

```diff
@@ -17,16 +17,16 @@
 # pylint: disable=unused-import
 import warnings
 
+from numba import cuda
+
 from merlin.core.has_gpu import HAS_GPU  # noqa pylint: disable=unused-import
 
 try:
     import psutil
 except ImportError:
     psutil = None
 
-try:
-    from numba import cuda
-except ImportError:
+if not cuda.is_available():
```

Comment:
this might not be a problem with this function. However, I recall we had an issue once where we had `cuda.gpus.lst` in the global scope and that caused some problems with Dask-CUDA https://github.com/NVIDIA-Merlin/core/issues/144 It might be worth double checking this won't have a similar side effect?